src/dispatch/builder.rs

```diff
@@ -272,11 +350,18 @@ impl<'a, 'b> DispatcherBuilder<'a, 'b> {
     pub fn build(self) -> Dispatcher<'a, 'b> {
         use crate::dispatch::dispatcher::new_dispatcher;
 
+        #[cfg(feature = "parallel")]
+        self.thread_pool
+            .write()
+            .unwrap()
+            .0
+            .get_or_insert(Self::create_thread_pool());
```

Comment:
btw should these be `get_or_insert_with` so it doesn't create thread pool if there is already something in the `Option`?

Reply:
Correct, thanks!